UnityProject/Assets/Scripts/Systems/Chemistry/Effects/ChemExplosion.cs

```diff
@@ -17,6 +17,9 @@ public class ChemExplosion : Chemistry.Effect
 		[Tooltip("Multiplier applied to final strength calculation")]
 		[SerializeField] private float potency = 1;
 
+		[Tooltip("Is explosion is actually EMP")]
+		[SerializeField] private bool isEMP = false;
```

Comment:
I'm not a fan of adding bools to scripts to make them behave like something else. This looks like it should have an abstract parent with generic explosion code and children that extend it

Reply:
That said, I can consider that refactor to be out of the scope of this particular pr if you don't feel like doing it. I just don't want to see "isFlashbang" bool everywhere in explosion code when that's added. It needs to be refactored before thinking on adding more.

Reply:
I might do something about it in separate PR but for now I don't really want to touch it